crates/flux-refineck/src/checker.rs

```diff
@@ -122,6 +122,19 @@ enum Guard {
     Match(Place, VariantIdx),
 }
 
+/// TODO: hack to prune the "extra" arm for `FluxExternEnum` (see option00.rs) which would, otherwise,
+/// generate a spurious `match` arm (i.e. `successor`) for the `FluxExternEnumFake` variant.
+fn adt_def_num_variants(adt_def: &AdtDef) -> usize {
+    let def_id = adt_def.did();
+    let is_extern_enum = format!("{def_id:?}").contains("__FluxExternEnum");
+    let num_variants = adt_def.variants().len();
+    if is_extern_enum {
+        num_variants - 1
+    } else {
+        num_variants
+    }
+}
```

Comment:
This should be fixed somewhere else. I believe the underlying problem is that we are creating the `AdtDef` in [`conv::adt_def_for_enum`](https://github.com/flux-rs/flux/blob/867519a6bc3fc7285ce36f82e3e19dee2b27bd2d/crates/flux-fhir-analysis/src/conv.rs#L220) with the fake enum rustc `AdtDef` instead of the real one. 

`adt_def_for_enum` should be aware of extern specs and adjust accordingly. Note that `adt_def_for_struct` has the same issue but doesn't cause problems because extern structs are always opaque.

Ideally, we could encode with an enum whether the spec is local or external. Then we could make the definition of `fhir::EnumDef` something like

```rust
enum LocalOrExtern {
    Local(OwnerId),
    Extern(OwnerId, DefId)
}

enum EnumDef {
    owner_id: LocalOrExtern,
    ...
 }
```

Reply:
Instead of the above, which messes up the `owner_id`, I just (equivalently) added an extra `extern_id: Option<DefId>` field to the struct which tracks if the spec is external.

```rust
#[derive(Debug)]
pub struct EnumDef {
    pub owner_id: OwnerId,
    pub params: Vec<RefineParam>,
    pub variants: Vec<VariantDef>,
    pub invariants: Vec<Expr>,
    /// Whether this is an extern_spec for some other enum
    pub extern_id: Option<DefId>,
}
```